wandb/sdk/wandb_run.py

```diff
@@ -2232,13 +2233,19 @@ def _public_api(self) -> PublicApi:
     # TODO(jhr): annotate this
     def _assert_can_log_artifact(self, artifact) -> None:  # type: ignore
         if not self._settings._offline:
-            public_api = self._public_api()
-            expected_type = public.Artifact.expected_type(
-                public_api.client,
-                artifact.name,
-                public_api.settings["entity"],
-                public_api.settings["project"],
-            )
+            try:
+                public_api = self._public_api()
+                expected_type = public.Artifact.expected_type(
+                    public_api.client,
+                    artifact.name,
+                    public_api.settings["entity"],
+                    public_api.settings["project"],
+                )
+            except requests.exceptions.ConnectionError:
```

Comment:
This would handle the present failure mode but I think we probably also want to do the same for timeouts, and basically anything other than the server explicitly saying "no, this isn't ok"

Reply:
I updated to a more general error class